documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+You can supply a custom callback function via `on` method of the `device` instance. It is recommended to use lifecycle
+events callbacks to help determine the state of the MQTT client during operation.
+
+The v2 SDK defines its own set of lifecycle events: *AttemptingConnect*, *ConnectionSuccess*, *ConnectionFailure*, *Disconnect*,
+and *Stopped*. For more information, refer to the [MQTT5 user guide](https://github.com/awslabs/aws-crt-nodejs/blob/main/MQTT5-UserGuide.md#client-events).
+
+#### Example of setting lifecycle events in the v1 SDK
+
+```typescript
+device
+    .on('connect', function() {
+        console.log('connect');
+    });
+device
+    .on('close', function() {
+        console.log('close');
+    });
+device
+    .on('reconnect', function() {
+        console.log('reconnect');
+    });
+device
+    .on('offline', function() {
+        console.log('offline');
+    });
+device
+    .on('error', function(error) {
+        console.log('error', error);
+    });
+```
+
+#### Example of setting lifecycle events in the v2 SDK
+
+```typescript
+client.on('error', (error: ICrtError) => {
+    console.log("Error event: " + error.toString());
+});
+
+client.on('attemptingConnect', (eventData: mqtt5.AttemptingConnectEvent) => {
+    console.log("Attempting Connect event");
+});
+
+client.on('connectionSuccess', (eventData: mqtt5.ConnectionSuccessEvent) => {
+    console.log("Connection Success event");
+    console.log ("Connack: " + JSON.stringify(eventData.connack));
+    console.log ("Settings: " + JSON.stringify(eventData.settings));
+});
+
+client.on('connectionFailure', (eventData: mqtt5.ConnectionFailureEvent) => {
+    console.log("Connection failure event: " + eventData.error.toString());
+    if (eventData.connack) {
+        console.log ("Connack: " + JSON.stringify(eventData.connack));
+    }
+});
+
+client.on('disconnection', (eventData: mqtt5.DisconnectionEvent) => {
+    console.log("Disconnection event: " + eventData.error.toString());
+    if (eventData.disconnect !== undefined) {
+        console.log('Disconnect packet: ' + JSON.stringify(eventData.disconnect));
+    }
+});
+
+client.on('stopped', (eventData: mqtt5.StoppedEvent) => {
+    console.log("Stopped event");
+});
+```
+
+
+### Publish
+
+The `publish` operation in the v1 SDK takes a topic and a message directly as parameters. The result of the `publish`
+operation in the v1 SDK is reported via a callback. If you try to publish to a topic that is not allowed by a policy,
+AWS IoT Core service will close the connection.
+
+In the v2 SDK, the [publish](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/mqtt5.Mqtt5Client.html#publish)
+operation takes a description of the PUBLISH packet you wish to send and returns a promise of polymorphic value.
+
+> [!NOTE]
+> If you try to publish with the v2 MQTT5 client to a topic that is not allowed by a policy, you do not get the connection
```

Comment:
Trivial: "If you publish using the v2 MQTT5 client to a topic not allowed by a policy, AWS IoT Core service will not close the connection and will instead send a PUBACK with the "Not authorized" reason code."